src/App.tsx

```diff
@@ -1,157 +1,96 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Filter, Todo } from './types/Todo';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { ErrorMessage } from './Components/ErrorMessage';
+import { callbacks } from './localStorage/localStorage';
+import { ContextProvider } from './Components/ContextProvider';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const savedTodos = localStorage.getItem('todos');
+
+    return savedTodos ? JSON.parse(savedTodos) : [];
+  });
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filter, setFilter] = useState(Filter.All);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setTodos(callbacks.getTodos());
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.Completed:
+        return todos.filter(todo => todo.completed);
+      case Filter.All:
+      default:
+        return todos;
+    }
+  }, [todos, filter]);
+
+  const deleteTodo = useCallback(
+    (todoId: number) => {
+      setErrorMessage('');
+
+      const updatedTodos = todos.filter(todo => todo.id !== todoId);
+
+      setTodos(updatedTodos);
+
+      setTimeout(() => {
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      }, 0);
+
+      return callbacks.setTodos(updatedTodos);
+    },
+    [todos],
+  );
```

Comment:
all this logic should be in context, if you has states and logic in components why you need context?)